tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -671,7 +671,10 @@ def __init__(self, filename, mode):
             )
         self.filename = compat.as_bytes(filename)
         self.fs = get_filesystem(self.filename)
-        self.fs_supports_append = hasattr(self.fs, "append")
+        if _get_fs_class(self.filename) == AzureBlobFileSystem:
```

Comment:
It's a bit odd that we would call get_filesystem, which calls fsspec.get_filesystem_class transitively, and then we call fsspec.get_filesystem_class again.

Can we remove the duplication?

Perhaps get_filesystem could return a tuple, where first entry is the filesystem (like _FSSPEC_FILESYSTEM) and the second entry is the filesystem class (if any, like AzureBlobFileSystem).